protocols/raft/src/main/java/io/atomix/protocols/raft/proxy/impl/RaftProxyManager.java

```diff
@@ -229,6 +231,36 @@ public void resetConnections(MemberId leader, Collection<MemberId> servers) {
     return future;
   }
 
+  /**
+   * Resets indexes for all sessions.
+   */
+  private synchronized void resetAllIndexes() {
```

Comment:
After a leader change, a new leader will have stale information about the client’s sessions. Sending another keep-alive to the new leader causes indexes and sequence numbers to be reset on the new leader to ensure the sessions can more quickly make progress after the leader change. Without this change, often leaders have to wait for the next scheduled keep-alive from a client before they can continue accepting the client’s commands in sequential order.